bson/son.py

```diff
@@ -80,21 +80,16 @@ def __iter__(self):
     def has_key(self, key):
         return key in self.__keys
 
-    # third level takes advantage of second level definitions
-    def iteritems(self):
-        for k in self:
-            yield (k, self[k])
-
     def iterkeys(self):
         return self.__iter__()
 
     # fourth level uses definitions from lower levels
     def itervalues(self):
-        for _, v in self.iteritems():
+        for _, v in self.items():
             yield v
 
     def values(self):
-        return [v for _, v in self.iteritems()]
+        return [v for _, v in self.items()]
 
     def items(self):
         return [(key, self[key]) for key in self]
```

Comment:
Using items instead of iteritems is going to be a performance problem because items() returns a fully inflated list. If we want to make this change we have to avoid that.

Reply:
Done.